app/chat_search/ai_services/service.py

```diff
@@ -0,0 +1,134 @@
+import re
+from typing import Any
+from .initialized_huggingface_embeddings import InitializedHuggingFaceEmbeddings
+from langchain.embeddings.huggingface import HuggingFaceEmbeddings
+from langchain.document_loaders.csv_loader import CSVLoader
+from langchain.retrievers import BM25Retriever, EnsembleRetriever
+from langchain.vectorstores import FAISS
+from langchain.chains import RetrievalQA
+from pydantic import BaseModel
+
+from app.aimodels.gpt4all.ai_services.completion_inference import (
+    CompletionInference,
+    CompletionInferenceInputs,
+)
+from app.chat_search.ai_services.marco_rerank_retriever import MarcoRerankRetriever
+from app.core.errors import ValidationError
+from app.core.minio import download_pickled_object_from_minio
+from app.core.model_cache import MODEL_CACHE_BASEDIR
+
+from sqlalchemy.orm import Session
+from minio import Minio
+from app.aimodels.bertopic.crud import bertopic_embedding_pretrained
+
+from sample_data import CHAT_DATASET_1_PATH
```

Comment:
@erob123 TODO replace chat dataset with list of document uuids https://github.com/orgs/MIT-AI-Accelerator/projects/2/views/1?pane=issue&itemId=35691850